### PR TITLE
Allow authentication with Organization and User

### LINKED ISSUE #21: Disassociate user from ledger authentication
**Is your feature request related to a problem? Please describe.**
When you want to authenticate a new ledger, you must create a user as `current_user` is used to determine the `organization`.  When using the app solely for authentication, it may not make sense to create users.  You may only need organizations and ledger authentication.

**Describe the solution you'd like**
- Make it possible to authenticate a ledger without `current_user`
- Potentially add `organization` to `AuthToken` to signal that there will be no user session.  Or maybe create a new `LedgerAuthToken` table.

### PR DESCRIPTION
**Pull request by ryanwjackson**
Fix #21

### CODE REVIEW

File: app/controllers/ui_controller.rb [hunk 1]
```diff
@@ -9,8 +9,18 @@ class UIController < ApplicationController
...
 
+  def authorized_as_organization?
+    @authorized_as_organization ||= current_user.is_a?(Organization)
+  end
+
   def current_organization
-    @current_organization ||= current_user.try(:organizations).try(:first).try(:decorate)
+    @current_organization ||= begin
+      if authorized_as_organization?
+        current_user
```
Comment (mattgmarcus) [inline app/controllers/ui_controller.rb]: Should we decorate this organization here too?

File: config/locales/en.yml [hunk 2]
```diff
@@ -48,6 +48,7 @@ en:
       success: 'You have been successfully logged out.'
     dev_login:
       success: 'You have been successfully logged in as a developer.'
+    invalid_organization_token: 'Invalid organization token.'
```
Comment (mattgmarcus) [inline config/locales/en.yml]: Is this used anywhere? I may just be missing it

File: app/controllers/auth_tokens_controller.rb [hunk 3]
```diff
@@ -21,7 +21,7 @@ def index
   end
 
   def show
-    authenticate!
+    authenticate!(:token)
```
Comment (mattgmarcus) [inline app/controllers/auth_tokens_controller.rb]: It looks like you're passing an arg here but in the warden file, it doesn't take an arg

File: app/controllers/api/v1/auth_tokens_controller.rb [hunk 4]
```diff
@@ -6,7 +6,7 @@ class AuthTokensController < API::V1::BaseController
       before_action :set_user
 
       def create
-        Forms::AuthTokens::Create.new(user: @user).save
+        Forms::AuthTokens::Create.new(resource: @user).save
```
Comment (mattgmarcus) [inline app/controllers/api/v1/auth_tokens_controller.rb]: We should have a separate route for the option where you do `POST /api/v1/organizations/:organization_id/auth_tokens` right?

**Comment by ryanwjackson** [comment]
Closing this PR (see #21 for details).

**Pull request closed without merging**